hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/io/storage/row/HoodieRowCreateHandle.java

```diff
@@ -241,6 +242,9 @@ public WriteStatus close() throws IOException {
     stat.setTotalWriteBytes(fileSizeInBytes);
     stat.setFileSizeInBytes(fileSizeInBytes);
     stat.setTotalWriteErrors(writeStatus.getTotalErrorRecords());
+    for (Pair<HoodieRecordDelegate, Throwable> pair : writeStatus.getFailedRecords()) {
+      LOG.error("Failed to write {}", pair.getLeft(), pair.getRight());
+    }
```

Comment:
Is there any possibility we have too many records to print so the logs are overwhelmed.

Reply:
There's low possibility as Hudi doesn't store all the exception in `writeStatus.getFailedRecords()`. By default, 10% of the errors are stored and the percentage is configurable via `hoodie.memory.writestatus.failure.fraction`. Note that the first error is always stored.

Reply:
So do you think we just print out all the errors or just the first one?

Reply:
For troubleshooting, I prefer printing out all the errors.